### DISCUSSION TITLE
PDF Viewer for Sources

### DISCUSSION THREAD

**Discussion by johanneskopton** [Zettlr/Zettlr#2858] [Ideas]
Hey lovely zettlr people,

Im quite new here, but I had the following idea (might be an old hat or completely useless).

I found that most of my sources are in PDF format (academic papers, ..). And I cite them using the citekey and my Zotero-generated .bib-file. However, if I stumble upon such a citation in my zettelkasten and want to read it, I have to switch to Zotero and search for the entry and double-click on it.

This could be easier, since the global filepath of each attachement is put in to the .bib by Zotero. Wouldn't it make sense, if the bibliography entries in the sidebar would be clickable to open the pdf?

It would be nice if it would open in my default pdf viewer, but maybe it would be even nicer, if it would open in a new zettlr tab (thanks to pdf.js). Then there could even be a box linking back to the markdown file, that cited the respective page of the pdf source.

Same thing could be done for epub.

Im not too much of a web developer (and relatively busy at the moment) but I think I might try to implement something like that.
Would anyone find it useful?

Cheers
Johannes

**Comment by nathanlesage** [comment]
> This could be easier, since the global filepath of each attachement is put in to the .bib by Zotero.

You should be able to right-click a citation and select "Open", since Zettlr will then just open the corresponding PDF in your default PDF viewer.

In general, opening PDFs directly in Zettlr is a possibility, but has low priority due to other work right now.
